lib/puppet/util/checksums.rb

```diff
@@ -6,11 +6,14 @@
 module Puppet::Util::Checksums
   module_function
 
+  def valid_checksum_types
+    [:sha256, :sha256lite, :md5, :md5lite, :sha1, :sha1lite, :mtime, :ctime]
+  end
+
   # It's not a good idea to use some of these in some contexts: for example, I
   # wouldn't try bucketing a file using the :none checksum type.
   def known_checksum_types
-    [:sha256, :sha256lite, :md5, :md5lite, :sha1, :sha1lite,
-      :mtime, :ctime, :none]
+    self.preferred_checksum_types.push :none
```

Comment:
I think it exists instead of nil. I'm not really sure, except maybe it's required by the indirector so we know it's not just using the default. I'm not sure how you'd ever have :none without an error though.